### ISSUE TITLE
CupertinoPageRoute throws exception and Navigator is stuck

### ISSUE THREAD

**Issue by najeira** [flutter/flutter#28876]
## Steps to Reproduce

- Pop a route by pop gesture (a.k.a swipe back)
- Push a route immediately

In the sample application below, it is reproduced by the following steps:

- Push "Next"
- Swipe back
- Push "Next" immediately

```
import 'package:flutter/material.dart';
import 'package:flutter/cupertino.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Pop gesture"),
      ),
      body: RaisedButton(
        child: Text("Next"),
        onPressed: () {
          Navigator.push(context, CupertinoPageRoute(
            builder: (BuildContext context) {
              return Scaffold(
                backgroundColor: Colors.yellow,
                appBar: AppBar(title: Text("Second")),
              );
            },
          ));
        },
      ),
    );
  }
}
```

## Logs

```
I/flutter (30832): ══╡ EXCEPTION CAUGHT BY ANIMATION LIBRARY ╞═════════════════════════════════════════════════════════
I/flutter (30832): The following assertion was thrown while notifying status listeners for AnimationController:
I/flutter (30832): Cannot reuse a CupertinoPageRoute<dynamic> after disposing it.
I/flutter (30832): 'package:flutter/src/widgets/routes.dart': Failed assertion: line 217 pos 12:
I/flutter (30832): '!_transitionCompleter.isCompleted'
I/flutter (30832): 
I/flutter (30832): Either the assertion indicates an error in the framework itself, or we should provide substantially
I/flutter (30832): more information in this error message to help you determine and fix the underlying cause.
I/flutter (30832): In either case, please report this assertion by filing a bug on GitHub:
I/flutter (30832):   https://github.com/flutter/flutter/issues/new?template=BUG.md
I/flutter (30832): 
I/flutter (30832): When the exception was thrown, this was the stack:
I/flutter (30832): #2      TransitionRoute.didPopNext (package:flutter/src/widgets/routes.dart:217:12)
I/flutter (30832): #3      NavigatorState.pop (package:flutter/src/widgets/navigator.dart:1944:23)
I/flutter (30832): #4      _CupertinoBackGestureController._handleStatusChanged (package:flutter/src/cupertino/route.dart:655:17)
I/flutter (30832): #5      _AnimationController&Animation&AnimationEagerListenerMixin&AnimationLocalListenersMixin&AnimationLocalStatusListenersMixin.notifyStatusListeners (package:flutter/src/animation/listener_helpers.dart:192:19)
I/flutter (30832): #6      AnimationController._checkStatusChanged (package:flutter/src/animation/animation_controller.dart:736:7)
I/flutter (30832): #7      AnimationController._tick (package:flutter/src/animation/animation_controller.dart:752:5)
I/flutter (30832): #8      Ticker._tick (package:flutter/src/scheduler/ticker.dart:228:5)
I/flutter (30832): #9      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:988:15)
I/flutter (30832): #10     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleBeginFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:904:11)
I/flutter (30832): #11     __InternalLinkedHashMap&_HashVMBase&MapMixin&_LinkedHashMapMixin.forEach (dart:collection/runtime/libcompact_hash.dart:367:8)
I/flutter (30832): #12     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleBeginFrame (package:flutter/src/scheduler/binding.dart:902:17)
I/flutter (30832): #13     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._handleBeginFrame (package:flutter/src/scheduler/binding.dart:832:5)
I/flutter (30832): #17     _invoke1 (dart:ui/hooks.dart:223:10)
I/flutter (30832): #18     _beginFrame (dart:ui/hooks.dart:162:3)
I/flutter (30832): (elided 5 frames from class _AssertionError and package dart:async)
I/flutter (30832): 
I/flutter (30832): The AnimationController notifying status listeners was:
I/flutter (30832):   AnimationController#1892e(⏮ 0.000; paused; DISPOSED; for CupertinoPageRoute<dynamic>(null))
I/flutter (30832): ════════════════════════════════════════════════════════════════════════════════════════════════════
```

and 

```
I/flutter (30832): Another exception was thrown: 'package:flutter/src/widgets/navigator.dart': Failed assertion: line 1666 pos 12: '!_debugLocked': is not true.
```

```
[✓] Flutter (Channel unknown, v1.2.1, on Mac OS X 10.14.2 18C54, locale ja-JP)
    • Flutter version 1.2.1 at /Applications/flutter
    • Framework revision 8661d8aecd (3 weeks ago), 2019-02-14 19:19:53 -0800
    • Engine revision 3757390fa4
    • Dart version 2.1.2 (build 2.1.2-dev.0.0 0a7dcf17eb)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/najeira/Library/Android/sdk
    • Android NDK at /Users/najeira/Library/Android/sdk/ndk-bundle
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /Users/najeira/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/najeira/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.0, Build version 10A255
    • ios-deploy 2.0.0
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 33.0.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] IntelliJ IDEA Ultimate Edition (version 2018.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 31.1.1
    • Dart plugin version 181.4203.498

[✓] VS Code (version 1.31.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.24.0

[✓] Connected device (2 available)
    • Android SDK built for x86 • emulator-5554    • android-x86   • Android 8.0.0 (API 26) (emulator)
```

**Comment by HansMuller** [comment]
This looks like a bug fixed just yesterday - https://github.com/flutter/flutter/pull/28756

**Comment by najeira** [comment]
I confirmed that this was fixed with #28756

However, I found another problem, I will reproduce and open issue.
